cypress/e2e/invoice/invoice.spec.cy.js

```diff
@@ -0,0 +1,160 @@
+// const cypress = require("cypress")
+
+describe('invoice module', () => {
+
+    env: {
+      username: ""
+    }
+    beforeEach(()=>{
+    
+       // cy.loginWithApi()
+       // cy.visit(Cypress.env('baseUrl'))
+       
+       cy.loginFromUi()
+       cy.get('#productManagerBtn')
+       cy.get(':nth-child(3) > .nav-link').click()
+       cy.url().should('includes', '/invoices')
+       cy.get('.Create-btn')
+       cy.wait(4000)
+        
+    }) 
+   
+    
+    after(()=>{
+       // logout from UI
+       // cy.get('#productManagerBtn').click()
+
+       // logout by removing the token from local storage
+       //cy.logout() 
+      
+
+    })
+
+    // test Case:1 create an invoice wihtout a coupon
+    it('should create an invoice', () => {
+  
+       
+       
+        cy.contains(' Create Invoice ').click()
+
+        // wait to load the complete objects on the page
+        cy.wait(10000)
+        
+        // select a customer
+        cy.get(':nth-child(1) > .row > .col-xl-8 > .payment-value > .ng-select > .ng-select-container').type('+923244323448').click()
+        cy.get('.ng-option-label').click()
+
+        // checking the currency field has a value=Egyption Pound 
+        cy.get('.mat-select-min-line').should('contain','Egyptian Pound')
+        
+        // selecting a data from the calender
+        cy.get('.mat-datepicker-toggle-default-icon').click()
+       // cy.get('.mat-calendar-body > .ng-star-inserted > .mat-calendar-body-active').click()
+       
+       cy.get('#mat-input-1').then(($date) => {
+          
+          const selectedDate = $date
+        
+          cy.get('.mat-calendar-body > .ng-star-inserted > .mat-calendar-body-active')
+            .click()
+            .then(() => {
+              const selectingDate = $date
+              
+              // make sure it's what we expected
+              expect(selectingDate).to.eq(selectedDate)
+          
+            })
+            
+        })
+        
+
+
+        // adding item name
+
+        cy.get('#mat-input-2').type('Item')
+        cy.get('#mat-input-2').should('have.value', 'Item').and('be.visible')
```

Comment:
@AmnaShahid96 try to use `cy.get().contains()` instead.
And the assertion for being visible here doesn't make any sense. It has to be visible since you added value to it one line above.